### PR TITLE
bazel: Make use of new incremental pip installer

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: bazel: Make use of new incremental pip installer
Additional Description:

this rationalizes the pip dependencies down to ~1 requirements file (there are a few others left for non bazel requirements atm)

also removes dep comments, as previously requested by @moderation 

Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
8( this seems to be blocked by https://github.com/bazelbuild/rules_python/issues/527

**Comment by phlax** [comment]
unblocked by using upstream main/HEAD

File: ci/do_ci.sh [hunk 1]
```diff
@@ -46,6 +46,7 @@ function collect_build_profile() {
 function bazel_with_collection() {
   local failed_logs
   declare -r BAZEL_OUTPUT="${ENVOY_SRCDIR}"/bazel.output.txt
+  echo "bazel $*"
```
Comment (phlax) [inline ci/do_ci.sh:49]: remove

File: ci/do_ci.sh [hunk 2]
```diff
@@ -330,7 +331,7 @@ elif [[ "$CI_TARGET" == "bazel.compile_time_options" ]]; then
     "--test_env=ENVOY_HAS_EXTRA_EXTENSIONS=true")
 
   ENVOY_STDLIB="${ENVOY_STDLIB:-libstdc++}"
-  setup_clang_toolchain
+  # setup_clang_toolchain
```
Comment (phlax) [inline ci/do_ci.sh:334]: revert

File: ci/do_ci.sh [hunk 3]
```diff
@@ -339,7 +340,8 @@ elif [[ "$CI_TARGET" == "bazel.compile_time_options" ]]; then
 
   # Building all the dependencies from scratch to link them against libc++.
   echo "Building and testing with wasm=wamr: ${TEST_TARGETS[*]}"
-  bazel_with_collection test "${BAZEL_BUILD_OPTIONS[@]}" --define wasm=wamr "${COMPILE_TIME_OPTIONS[@]}" -c dbg "${TEST_TARGETS[@]}" --test_tag_filters=-nofips --build_tests_only
+  bazel_with_collection test "${BAZEL_BUILD_OPTIONS[@]}" --define wasm=wamr "${COMPILE_TIME_OPTIONS[@]}" -c dbg "${TEST_TARGETS[@]}" --test_tag_filters=-nofips --build_tests_only --sandbox_debug
+  exit 0
```
Comment (phlax) [inline ci/do_ci.sh:344]: and this

**Comment by phlax** [comment]
following up offline conversation with @htuch - this currently removes the requirements.txt from the kafka extension

we may, for purposes of code ownership, want to keep some requirements.txt files separate - like, eg, the kafka one

mho is that we (want to) maintain the deps across the repo, and that doesnt tend to be so related to codeownership, more dependency shepherding, so for the sake of making the latter easier, the less files the better

File: tools/code_format/requirements.txt [hunk 4]
```diff
@@ -1,36 +0,0 @@
-#
```
Comment (phlax) [inline tools/code_format/requirements.txt:1]: removing this will break non-bazel code_format atm

**Review by mattklein123 — DISMISSED** [review]

**Review by htuch — APPROVED** [review]
LGTM, but I do remain a little uneasy at the lack of versioning in `requirements.in`. I'm going to go with what you said in your other review and accept that we probably don't care most of the time and when we do, we have the option to pin.

**Comment by phlax** [comment]
@mattklein123 i had to readd the requirements file for code_format

**Comment by phlax** [comment]
> I'm going to go with what you said in your other review and accept that we probably don't care most of the time and when we do, we have the option to pin.

yep, we can

not sure if you saw my other comment about cross-platform compilation of the lock file

atm pip-compile doesnt do the right thing on this, so if you want eg a package only on windows, you need to add it without any predicates, and then add in bazel so its only loaded in win

**Review by mattklein123 — APPROVED** [review]

**Pull request merged**